docs/ParameterValidation.md

```diff
@@ -0,0 +1,174 @@
+# Helium Parameter Validation Documentation
+
+This goals of this document are to define valid/invalid API and Query parameters and provide specifications on resulting behavior of the Helium API.
+
+## Common Status code rules followed for all APIs
+
+### Searches
+
+- Valid input returns 200 with JSON Array [{},{}...]
+- Valid input with no results returns 200 with empty array
+- Invalid input returns 400 text/plain with Invalid error response
+
+### Single Reads
+
+- Valid single reads returns 200 with JSON document
+- Valid single reads with no results returns 404 with No {entity} found error response
+- Invalid single reads like /api/movies/tt1234 , /api/actorid/nm1234 returns 400 with Invalid error response
+
+### Multiple Error Scenarios
+
+- Fail on first error
+
+## Movies
+
+### API /api/movies
+
+### Movies Query Parameters
+
+- Name: query (search)
+- Description: This a search query string on the movie title or actor name field
+- Type: string
+- Parameter Validation:
+  - Valid input: query length >= 3 non-whitespace characters
+    - Status: 200
+    - Content-Type: application/json
+    - Response Body: JSON Array of movies filtered by search query on title or actor name field as [{}, {},..], empty array when no results
+  - Invalid input: query length < 3 characters
+    - Status: 400
+    - Content-Type: text/plain
+    - Response Body: Invalid query parameter
+
+- Name: year
+- Type: integer
+- Parameter Validation:
+  - Valid input range: [1874 , currentYear+5]
+    - Status: 200
+    - Content-Type: application/json
+    - Response Body: JSON Array of movies filtered by year as [{}, {},..], empty array when no results
+  - Invalid input - input that does not parse or is out of range
+    - Status: 400
+    - Content-Type: text/plain
+    - Response Body: Invalid year parameter
+
+- Name: rating
+- Type: double
+- Parameter Validation:
+  - Valid input range: [0.0, 10.0]
+  - Round off more than single decimal to a single decimal
+    - Example: round off 7.253896 to 7.3
```

Comment:
When we talked I think we said that we'd take whatever the round function for the language did by default.  However, here it sounds like we will always round up at 5.  I think that is okay and gives good consistency, but let's call it out if it is a rule.  Round to a single digit and round up at 5.

Reply:
updated with "based on round function for the language"

Reply:
I don't see the update when I review in github web ... what am I missing?